core/services/keystore/eth_test.go

```diff
@@ -27,18 +29,20 @@ import (
 func Test_EthKeyStore(t *testing.T) {
 	t.Parallel()
 
-	db := pgtest.NewSqlxDB(t)
-	cfg := configtest.NewTestGeneralConfig(t)
+	db := sqlutil.WrapDataSource(pgtest.NewSqlxDB(t), logger.Test(t), sqlutil.TimeoutHook(func() time.Duration {
```

Comment:
Why wrap `pgtest.NewSqlxDb(t)`?

Reply:
Just mimicking production. This is where I debugged the issue.

Reply:
it's worth a code comment because it's not obvious.

or should the test db impl be changed to return the wrapper?

Reply:
Yeah, in the final PR #12456 I updated cltest at least to use a timeout wrapper. We could consider updating the test funcs to return a wrapped DataSource instead of raw sqlx most places :shrug: